carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/RecruitmentServiceHost.kt

```diff
@@ -124,57 +125,153 @@ class RecruitmentServiceHost(
+     *  - not all necessary participant roles part of the study have been assigned a participant
+     * @throws IllegalStateException when group has already been deployed.
+     */
+    override suspend fun inviteParticipantGroup( studyId: UUID, groupId: UUID ): ParticipantGroupStatus
+    {
+        val (recruitment, participantGroup) = getRecruitmentAndGroupOrThrow( studyId, groupId )
+        check( !participantGroup.isDeployed ) { "Participant group has already been deployed." }
 
-        // Create study deployment, which sends out invitations.
         // TODO: If the repository gets updated but `createStudyDeployment` fails, state will be inconsistent.
         //  This should use eventual consistency: https://github.com/imotions/carp.core-kotlin/issues/295
-        //  And probably be separate requests: https://github.com/imotions/carp.core-kotlin/issues/319
+
+        val (protocol, invitations) = recruitment.createInvitations( participantGroup.roleAssignments )
         val deploymentStatus = deploymentService.createStudyDeployment( participantGroup.id, protocol, invitations )
 
+        // Mark participant group as deployed.
+        participantGroup.markAsDeployed()
+        participantRepository.updateRecruitment( recruitment )
         return recruitment.getParticipantGroupStatus( deploymentStatus )
     }
 
     /**
-     * Get the status of all deployed participant groups in the study with the specified [studyId].
+     * Get the status of all participant groups in the study with the specified [studyId].
      *
      * @throws IllegalArgumentException when a study with [studyId] does not exist.
      */
     override suspend fun getParticipantGroupStatusList( studyId: UUID ): List<ParticipantGroupStatus>
     {
         val recruitment: Recruitment = getRecruitmentOrThrow( studyId )
 
+        // Get status list for staged participant groups.
+        val stagedParticipantGroups = recruitment.participantGroups
+            .filter { !it.value.isDeployed }
+            .values
+            .map { recruitment.getParticipantGroupStatus( it ) }
+
         // Get study deployment status list.
-        val studyDeploymentIds = recruitment.participantGroups.keys
+        val deployedStudyDeploymentIds = recruitment.participantGroups.filter { it.value.isDeployed }.keys
+
+        // Get status list for deployed participant groups.
         val studyDeploymentStatusList: List<StudyDeploymentStatus> =
-            if ( studyDeploymentIds.isEmpty() ) emptyList()
-            else deploymentService.getStudyDeploymentStatusList( studyDeploymentIds )
+            if ( deployedStudyDeploymentIds.isEmpty() ) emptyList()
+            else deploymentService.getStudyDeploymentStatusList( deployedStudyDeploymentIds )
 
-        return studyDeploymentStatusList.map { recruitment.getParticipantGroupStatus( it ) }
+        return stagedParticipantGroups + studyDeploymentStatusList.map { recruitment.getParticipantGroupStatus( it ) }
```

Comment:
For consistency, move the `.map { recruitment.getParticipantGroupStatus( it ) }` to the "Get status list for deployed participant groups" block.